### ISSUE TITLE
potential.sampling

### ISSUE THREAD

**Issue by CGao-STEM** [jacobjma/abTEM#52]
Dear Jacobjma

When I set sampling number in potential function, as shown below. The real space sampling x and y are slightly different.
I intend to simulate 4d datasets, in which the reciprocal space sampling kx and ky are equal.  May I ask how to make sampling x and y are equal to each other?
![image](https://user-images.githubusercontent.com/59644454/132710371-98f90ee7-49a3-46cd-8a58-0230256a1aef.png)

Chuang

**Comment by jacobjma** [comment]
The potential sampling is different from the detector sampling. The potential sampling is a simulation parameter without any real-world equivalent.

The `PixelatedDetector` allows the keyword `resample`, you can set this to the physical pixel size in mrad. Let me know if this works.

**Comment by CGao-STEM** [comment]
I set the potential function as
![image](https://user-images.githubusercontent.com/59644454/132759781-a4c85b07-9859-4bca-a2f0-1cc56eb8c104.png)
Then set probe function as
![image](https://user-images.githubusercontent.com/59644454/132759930-3647e3cc-b46b-44b1-b7ec-ecc193ee8972.png)
Finally, set pixelateddtector function as
![image](https://user-images.githubusercontent.com/59644454/132760046-735e4e20-4816-41e8-9b91-01fb75ef6bc5.png)

The result is shown as
![image](https://user-images.githubusercontent.com/59644454/132760157-8214351e-59f9-454d-8df7-5a9dc770f461.png)

I am wondering if I set resample in pixelated detector, do I need to change probe_grid_match(potential)? I guess there are 2 sampling in the detector space, which may cause some problems. Which function should I change to make resampling number in pixelateddetector function valid?
